x-pack/plugins/reporting/public/panel_actions/get_csv_panel_action.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import dateMath from '@elastic/datemath';
+import { i18n } from '@kbn/i18n';
+
+import { ContextMenuAction, ContextMenuActionsRegistryProvider } from 'ui/embeddable';
+import { PanelActionAPI } from 'ui/embeddable/context_menu_actions/types';
+import { kfetch } from 'ui/kfetch';
+import { toastNotifications } from 'ui/notify';
+import chrome from 'ui/chrome';
+import { API_BASE_URL_V1 } from '../../common/constants';
+
+const API_BASE_URL = `${API_BASE_URL_V1}/generate/immediate/csv/saved-object`;
+
+class GetCsvReportPanelAction extends ContextMenuAction {
+  constructor() {
+    super(
+      {
+        displayName: i18n.translate('xpack.reporting.dashboard.downloadCsvPanelTitle', {
+          defaultMessage: 'Download CSV',
+        }),
+        id: 'downloadCsvReport',
+        parentPanelId: 'mainMenu',
+      },
+      {
+        icon: 'document',
+      }
+    );
+  }
+
+  public async generateJobParams({ searchEmbeddable }: { searchEmbeddable: any }) {
+    const adapters = searchEmbeddable.getInspectorAdapters();
+    if (!adapters) {
+      return {};
+    }
+
+    if (adapters.requests.requests.length === 0) {
+      return {};
+    }
+
+    return searchEmbeddable.searchScope.searchSource.getSearchRequestBody();
```

Comment:
It looks like it will be important to "clean" this body object before sending it through kFetch. I have noticed it has a lot of fields that are only used by Kibana UI, and we only need fields that are going to be used for the ES query.

Let's see if we can get it to strictly confirm to this `JobParams` shape.

There is a type for this for server-side code: https://github.com/elastic/kibana/pull/34571/files#diff-549322e4a9f349a6c9ebc3ed9f38f603R37

I intended to rename that type to `JobParamPostPayload ` in https://github.com/elastic/kibana/pull/35038